### ISSUE TITLE
Regarding PMF calculation with error bar (bootsrapping)

### ISSUE THREAD

**Issue by SoumyaCYZ** [dnlbauer/WHAM#17]
Hope you are fine and doing well. I have performed an umbrella sampling simulation using the center of mass distance (dz) between membrane and peptide. I have pulled the peptide 4.4 nm to 0nm. Windows are nicely overlapped. I have tried to calculate the PMF using the following command--> "wham --min 0.0 --max 4.4 --bins 101 --temperature 310 --file metadata -o pmf.out --bt 100 --seed 1234". Can I fix the bins=101 (the total number of windows=41)? How should we fix the bin, tolerance, iterations, and bt parameters for WHAM? In some PMF calculations, I have seen bins=bt. It will be a great help if you tell me the protocol in detail. Apart from this what is the unit of free energy in the outputfile?

**Comment by dnlbauer** [comment]
Seems like your question is not really related to the tool, but to the method itself. I don't see a problem in setting bins greater than the number of windows. You just have to make sure there are still "enough" samples in each bin if the number of bins  > number of windows.

See https://github.com/dnlbauer/WHAM/issues/16 for the free energy unit.

**Comment by SoumyaCYZ** [comment]
Thanks, sir for your suggestion. I have checked all of my COLVAR files. Each window is nicely overlapped. I have a small doubt. When I used 10 or 20 bootstrapping steps (bt) the error was approx +/- 1.5 kJ/mol.  But when I increased bt to 200  the error was +/- 0.3 kJ/mol. Is there any correct protocol to choose bt?

**Comment by dnlbauer** [comment]
In theory, more BT runs will improve the estimation. I would suggest to run it with an increasing number of BT steps and plot the approximated error with the steps. At some point, the error should not change with more steps anymore. That was the process I followed at least.

**Comment by SoumyaCYZ** [comment]
Yes sir I observed the same trend. After certain BT steps the errors are not changing. But in Grossfield's code bootstrapping broke because the very high free energies caused underflows on the probability, which in turn ruins the bootstrapping. I think the process involves a large energy barrier, bootstrap is not a good option.   So in order to check the convergence we should plot the free energy profiles  (for e.g. 0-10, 0-20,0-30, and 0-40 ns) and compare (block averaging).

**Comment by dnlbauer** [comment]
The two methods are serving different purposes. Bootstrapping is for quantifying the error, not for checking convergence since it takes random samples while convergence is happing along the time coordinate.

Yes, to check convergence, plotting blocks with increasing time intervals is a valid option.

The bootstrapping approach can be used _afterwards_ to quantify the uncertainty of the calculated energy from the simulation. However, for bootstrapping to give meaningful results, you should first check for convergence and apply it to the converged trajectory.
